### ISSUE TITLE
String or Tuple Transform Parameters get converted to single number

### ISSUE THREAD

**Issue by romangrothausmann** [SuperElastix/SimpleElastix#119]
It seems string or tuple Transform Parameters get converted to a single number, e.g.:

```
stfx = sitk.TransformixImageFilter()

stfx.SetTransformParameterMap(sitk.GetDefaultParameterMap("translation"))
stfx.SetTransformParameter('Size', '4000 4000')
stfx.GetTransformParameter(0, 'Size')
('4000 4000',)
stfx.PrintParameterMap()
```
outputs 

```
(Size 4000)
````

Similar for e.g. `selx.SetInitialTransformParameterFileName("000127_2016_12_02_0104_s01_08.txt")` which yields `(InitialTransformParametersFileName 127.000000)` but the correct file is used by `selx/ElastixImageFilter`, so I guess this is a problem of `PrintParameterMap()`. 

It seems `stfx/TransformixImageFilter` is run with the output of `PrintParameterMap()`, at least it eats up all memory with the above settings.

**Comment by PertuyF** [comment]
Hi @romangrothausmann ,

I expect that `stfx.SetTransformParameterMap(sitk.GetDefaultParameterMap("translation"))` could create some problems as you are setting a `ParameterMap` in place of a `TransformParameterMap`. Technically it can work, but some parameters could cause conflicts. (any comments here @kaspermarstal ?)

I am not sure there is a preconfigured `TransformParameterMap`, so you can either re-use one from `selx.GetTransformParameterMap()`, or create all parameters in a blank `sitk.ParameterMap()`

The title speaks of `tuple` too, but your example only show the problem with `str`.
I believe it should work with `stfx.SetTransformParameter('Size', ('4000', '4000'))`, ElastiX expects a vector if you're to set a series of values for one parameter.

But I am more familiar with this syntax:
```python
P = stfx.GetTransformParameterMap()

P["parameter_name"] = ["value"]
P["other_parameter_name"] = ["value"] * 2  # if you want to repeat the value of course...
```

**Comment by kaspermarstal** [comment]
Yes there is a difference between "ParameterMaps" as returned by GetDefaultParameterMaps() and "TransformParameterMaps" which are actually returned by elastix after a registration. While they share the same type (a dict of vectors), ParameterMaps are exclusively for input for elastix and define registration configurations, and TransformParameterMaps are exclusively for transformix and defines only a transformation. 

GetDefaultParameterMaps() returns input to elastix.

**Comment by romangrothausmann** [comment]
Thanks @PertuyF for pointing out that a vector of strings is expected, which solves one issue. An example line like Yours would do well on the docs page for Parameter Maps (https://simpleelastix.readthedocs.io/ParameterMaps.html).

Concerning `SetInitialTransformParameterFileName("000127_2016_12_02_0104_s01_08.txt")`, how to correctly save the final Parameter Map that was used by `ElastixImageFilter` for a registration? With: `selx.WriteParameterFile(selx.GetTransformParameterMap(0), "transform.txt")` I get the afore mentioned `(InitialTransformParametersFileName 127.000000)`, i.e. the string gets converted to a number.

**Comment by PertuyF** [comment]
Does it produce the same result if you don't use the setter `SetInitialTransformParameterFileName()` but the python `setattr` syntax I mentioned?
I'm not sure whether the setter parses the string, or whether `WriteParameterFile()` does it.

**Comment by romangrothausmann** [comment]
If I use the setattr syntax for the `InitialTransformParametersFileName` it does not work at all, i.e. no initial transform is performed nor reported by selx: 

```
pM['InitialTransformParametersFileName']= ['000127_2016_12_02_0104_s01_08.txt']
print pM['InitialTransformParametersFileName']
('000127_2016_12_02_0104_s01_08.txt',)
selx.SetParameterMap(pM)
print selx.GetInitialTransformParameterFileName(),
' '
```

but with `SetInitialTransformParameterFileName(...)`, `GetInitialTransformParameterFileName()` reports the correct value and an initial transform is performed:

```
selx.SetInitialTransformParameterFileName('000127_2016_12_02_0104_s01_08.txt')
print selx.GetInitialTransformParameterFileName(),
'000127_2016_12_02_0104_s01_08.txt'
print selx.GetTransformParameterMap(0)['InitialTransformParametersFileName']
'000127_2016_12_02_0104_s01_08.txt'
```

So it seems `WriteParameterFile()` does the wrong parsing, and for some reason the setattr syntax for the `InitialTransformParametersFileName` does not work.

**Comment by kaspermarstal** [comment]
Ahh, I see. This is something I did not think about when the parser was written. Internally, `InitialTransformParametersFileName` is an index, but it is not converted to a file name when written to disk. This has to be implemented upstream in elastix. Will get back to you.

**Comment by kaspermarstal** [comment]
Fixed by SuperElastix/elastix@9ba16836e1a0642ecd2006a70ce3886b1fa4720e.

**Comment by romangrothausmann** [comment]
Hi @kaspermarstal, I just stumbled on this problem again, with [selx@bcef08a](https://github.com/SuperElastix/SimpleElastix/tree/bcef08a453ad8b3830df851e9ee86cf1aa1e1654.)
`selx.GetInitialTransformParameterFileName()` prints: `000459_17-534_0116_s01_01.txt`
but 
`selx.PrintParameterMap(selx.GetTransformParameterMap())` prints:  `(InitialTransformParametersFileName 459.000000)`

This became a problem when I tried to apply the transform found by selx with stfx (https://github.com/romangrothausmann/elastix_scripts/commit/3727098d9d6d726c8a39648e69999ea2f5758838#diff-0897a62892423c7e082bc3845849fc96R204), which became necessary when implementing @dzenanz idea to use otsu+distance_map as input to selx, and also wanting to transform the original moving image.

I checked if [9ba1683](https://github.com/SuperElastix/elastix/commit/9ba16836e1a0642ecd2006a70ce3886b1fa4720e) is an ancestor of 0ce6c24a2b, which seems to be the case. Did the modifications get lost in some (merge) commits in between?
